scalajslib/src/mill/scalajslib/api/ScalaJSApi.scala

```diff
@@ -107,3 +107,93 @@ object JsEnvConfig {
       autoExit: Boolean
   ) extends JsEnvConfig
 }
+
+class OutputPatterns private (
+    private[scalajslib] val jsFile: String,
+    private[scalajslib] val sourceMapFile: String,
+    private[scalajslib] val moduleName: String,
+    private[scalajslib] val jsFileURI: String,
+    private[scalajslib] val sourceMapURI: String
```

Comment:
Why are these private?

Reply:
As the `apply` and the `withX` methods are public, there is no reason to keep these `val`s non-public.

Reply:
Inspired by Scala.js. This configuration class is meant to be a "write-only", there is no usage on the reading side, that's why I think they make the values package-private [here](https://github.com/scala-js/scala-js/blob/1ec64710273cec8ba41da4aa4c94dfa97d3fb872/linker-interface/shared/src/main/scala/org/scalajs/linker/interface/OutputPatterns.scala#L31).

Reply:
The `apply` is private as well

Reply:
I just noticed the `apply` is private, too. Still, I think, if we can create it with some method e.g. `withJSFile` accepting a `String`, than there is sooner or later someone who will want to programmatically retrieve that `String`, e.g. to manipulate it slightly. 

If there is no other good reason (as we don't protect use from extra API maintenance here), we should not make it harder that necessary and make it public.

Reply:
Usually, I prefer to follow whatever the Scala.js plugin does, but what you said makes sense.
Made them public ✅